src/main/java/org/truffleruby/interop/OutgoingForeignCallNode.java

```diff
@@ -113,6 +123,14 @@ protected OutgoingNode createHelperNode(int argsLength) {
             return new CallOutgoingNode(argsLength);
         } else if (name.equals("new")) {
             return new NewOutgoingNode(argsLength);
+        } else if (name.equals("to_a") || name.equals("to_ary")) {
+            return new ToAOutgoingNode();
+        } else if (name.equals("respond_to?")) {
+            return new RespondToOutgoingNode();
+        } else if (name.equals("inspect")) {
+            return new InspectOutgoingNode();
+        } else if (name.equals("send")) {
```

Comment:
Right, it just seems a particularly likely problem.
Also, the conversion code in type.rb should definitely use `__send__` for consistency and because that's what is right for Ruby objects as well (`send` is only Kernel not BasicObject and it is overridden by Socket).